File: singularity/implementation.java/Math.java
/* Copyright 2018 ComdivByZero
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package o7;

import o7.O7;

public final class Math {

public static final double pi = 3.14159265358979323846;
public static final double e  = 2.71828182845904523536;

public static double sqrt(double x) {
	return java.lang.Math.sqrt(x);
}

public static double power(double x, double base) {
	return java.lang.Math.pow(x, base);
}

public static double exp(double x) {
	return java.lang.Math.exp(x);
}

public static double ln(double x) {
	return java.lang.Math.log(x);
}

public static double log(double x, double base) {
	final double l;
	if (base == 10.0) {
		l = java.lang.Math.log10(x);
	} else if (x == e) {
		l = java.lang.Math.log(x);
	} else {
		l = java.lang.Math.log(x) / java.lang.Math.log(base);
	}
	return l;
}

public static double round(double x) {
	return java.lang.Math.rint(x);
}

public static double sin(double x) {
	return java.lang.Math.sin(x);
}

public static double cos(double x) {
	return java.lang.Math.cos(x);
}

public static double tan(double x) {
	return java.lang.Math.tan(x);
}

public static double arcsin(double x) {
	return java.lang.Math.asin(x);
}

public static double arccos(double x) {
	return java.lang.Math.acos(x);
}

public static double arctan(double x) {
	return java.lang.Math.atan(x);
}

public static double arctan2(double x, double y) {
	return java.lang.Math.atan2(x, y);
}

public static double sinh(double x) {
	return java.lang.Math.sinh(x);
}

public static double cosh(double x) {
	return java.lang.Math.cosh(x);
}

public static double tanh(double x) {
	return java.lang.Math.tanh(x);
}

public static double arcsinh(double x) {
	O7.asrt(false);
	return 0.0;
}

public static double arccosh(double x) {
	O7.asrt(false);
	return 0.0;
}

public static double arctanh(double x) {
	O7.asrt(false);
	return 0.0;
}

}

File: singularity/implementation.js/Math.js
/* Copyright 2019 ComdivByZero
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var o7;
(function(o7) { 'use strict';

var module = {};
o7.export.Math = module;

module.pi = Math.PI;
module.e  = Math.E;

function sqrt(x) {
	/* TODO */
	return Math.sqrt(x);
}
module.sqrt = sqrt;

function power(x, base) {
	/* TODO */
	return Math.pow(x, base);
}
module.power = power;

function exp(x) {
	/* TODO */
	return Math.exp();
}
module.exp = exp;

function ln(x) {
	/* TODO */
	return Math.log(x);
}
module.ln = ln;

function log(x, base) {
	/* TODO */
	var d;
	if (base == 10) {
		d = Math.LN10;
	} else {
		d = Math.log(base);
	}
	return Math.log(x) / d;
}
module.log = log;

function round(x) {
	/* TODO */
	return Math.round(x);
}
module.round = round;

function sin(x) {
	/* TODO */
	return Math.sin(x);
}
module.sin = sin;

function cos(x) {
	/* TODO */
	return Math.cos(x);
}
module.cos = cos;

function tan(x) {
	/* TODO */
	return Math.tan(x);
}
module.tan = tan;

function arcsin(x) {
	/* TODO */
	return Math.asin(x);
}
module.arcsin = arcsin;

function arccos(x) {
	/* TODO */
	return Math.acos(x);
}
module.arccos = arccos;

function arctan(x) {
	/* TODO */
	return Math.atan(x);
}
module.arctan = arctan;

function arctan2(x, y) {
	/* TODO */
	return Math.atan2(x, y);
}
module.arctan2 = arctan2;

function sinh(x) {
	/* TODO */
	return Math.sinh(x);
}
module.sinh = sinh;

function cosh(x) {
	/* TODO */
	return Math.cosh(x);
}
module.cosh = cosh;

function tanh(x) {
	/* TODO */
	return Math.tanh(x);
}
module.tanh = tanh;

function arcsinh(x) {
	/* TODO */
	return Math.asinh(x);
}
module.arcsinh = arcsinh;

function arccosh(x) {
	/* TODO */
	return Math.acosh(x);
}
module.arccosh = arccosh;

function arctanh(x) {
	/* TODO */
	return Math.atanh(x);
}
module.arctanh = arctanh;

return module;
})(o7 || (o7 = {}));

